central/graphql/resolvers/node_vulnerabilities.go

```diff
@@ -80,8 +143,243 @@ func (resolver *Resolver) NodeVulnCounter(ctx context.Context, args RawQuery) (*
 		query := withNodeCveTypeFiltering(args.String())
 		return resolver.vulnCounterV2(ctx, RawQuery{Query: &query})
 	}
-	// TODO : Add postgres support
-	return nil, errors.New("Resolver NodeVulnCounter does not support postgres yet")
+
+	if err := readNodes(ctx); err != nil {
+		return nil, err
+	}
+	query, err := args.AsV1QueryOrEmpty(search.ExcludeFieldLabel(search.Fixable))
+	if err != nil {
+		return nil, err
+	}
+	vulnLoader, err := loaders.GetNodeCVELoader(ctx)
+	if err != nil {
+		return nil, err
+	}
+	query = tryUnsuppressedQuery(query)
+	fixableVulnsQuery := search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery())
+	fixableVulns, err := vulnLoader.FromQuery(ctx, fixableVulnsQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	unFixableVulnsQuery := search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, false).ProtoQuery())
+	unFixableCVEs, err := vulnLoader.FromQuery(ctx, unFixableVulnsQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	return mapNodeCVEsToVulnerabilityCounter(fixableVulns, unFixableCVEs), nil
+}
+
+// CreatedAt is the time a node CVE first seen in the system
+func (resolver *nodeCVEResolver) CreatedAt(ctx context.Context) (*graphql.Time, error) {
+	value := resolver.data.GetCveBaseInfo().GetCreatedAt()
+	return timestamp(value)
+}
+
+// CVE name of the node CVE
+func (resolver *nodeCVEResolver) CVE(ctx context.Context) string {
+	return resolver.data.GetCveBaseInfo().GetCve()
+}
+
+// EnvImpact is the fraction of nodes that contain the nodeCVE
+func (resolver *nodeCVEResolver) EnvImpact(ctx context.Context) (float64, error) {
+	nodeLoader, err := loaders.GetNodeLoader(ctx)
+	if err != nil {
+		return 0, err
+	}
+	numNodes, err := nodeLoader.CountAll(ctx)
+	if err != nil || numNodes == 0 {
+		return 0, err
+	}
+
+	query := resolver.getNodeCVEQuery()
+	numNodesWithCVE, err := nodeLoader.CountFromQuery(ctx, query)
+	if err != nil || numNodesWithCVE == 0 {
+		return 0, err
+	}
+	return float64(numNodesWithCVE) / float64(numNodes), nil
+}
+
+// FixedByVersion returns the version of the parent component that removes this CVE
+func (resolver *nodeCVEResolver) FixedByVersion(_ context.Context) (string, error) {
+	scope, hasScope := scoped.GetScope(resolver.ctx)
+	if !hasScope {
+		return "", nil
+	}
+	if scope.Level != v1.SearchCategory_NODE_COMPONENTS {
+		return "", nil
+	}
+	edgeID := postgres.IDFromPks([]string{scope.ID, resolver.data.GetId()})
```

Comment:
@mandar mentioned that we can and should remove the dependency on this function, check out #2253 to see how I handled this function